contracts/bridge/wrappers/L2BridgeZap.sol

```diff
@@ -10,63 +10,53 @@ import "../interfaces/IWETH9.sol";
 contract L2BridgeZap {
     using SafeERC20 for IERC20;
 
-    ISynapseBridge synapseBridge;
+    ISynapseBridge public immutable synapseBridge;
+    // solhint-disable-next-line
     address payable public immutable WETH_ADDRESS;
 
     mapping(address => address) public swapMap;
     mapping(address => IERC20[]) public swapTokensMap;
 
-    uint256 constant MAX_UINT256 = 2**256 - 1;
+    uint256 private constant MAX_UINT256 = 2**256 - 1;
 
     constructor(
         address payable _wethAddress,
-        address _swapOne,
-        address tokenOne,
-        address _swapTwo,
-        address tokenTwo,
+        address[] memory _swaps,
+        address[] memory _tokens,
         ISynapseBridge _synapseBridge
     ) public {
+        require(_swaps.length == _tokens.length, "Arrays length differs");
         WETH_ADDRESS = _wethAddress;
         synapseBridge = _synapseBridge;
-        swapMap[tokenOne] = _swapOne;
-        swapMap[tokenTwo] = _swapTwo;
         if (_wethAddress != address(0)) {
-            IERC20(_wethAddress).safeIncreaseAllowance(address(_synapseBridge), MAX_UINT256);
+            IERC20(_wethAddress).safeIncreaseAllowance(
```

Comment:
I've encountered an issue on Optimism specifically where if WETH is both a token being approved via `_saveSwap` and being approved in constructor, hits an approval overflow limit. Fixed this by removing the initial approve and only approving it once. Not an issue anywhere else, just something to keep in mind.

Reply:
Resolved in [86a9b68](https://github.com/synapsecns/synapse-contracts/pull/116/commits/86a9b6897f7aae59e6739fd13e3b04dc631fe512)